channels/applicationinsights-channel-js/Tests/Unit/src/Sender.tests.ts

```diff
@@ -48,13 +66,506 @@ export class SenderTests extends AITestClass {
+                let payload = ["payload1", "payload2", "payload3", "payload4", "payload5", "payload6"];
+                arrForEach(payload, (val) =>{
+                    sessionBuffer.enqueue(val);
+                });
+                QUnit.assert.deepEqual(this._getBuffer(BUFFER_KEY, logger), payload, "session storage buffer is set");
+                sessionBuffer.deepCopy(arrBuffer);
+                QUnit.assert.deepEqual(arrBuffer.getItems(), payload, "payload should be same");
+                sessionBuffer.enqueue("payload");
+                QUnit.assert.deepEqual(sessionBuffer.getItems().length, 1, "sessionBuffer length");
+                QUnit.assert.deepEqual(this._getBuffer(BUFFER_KEY, logger), ["payload"], "session storage buffer is set");
+                QUnit.assert.deepEqual(arrBuffer.getItems().length, 6, "copy is deep copy");
+
+                utlSetSessionStorage(logger, BUFFER_KEY,JSON.stringify([]));
+            }
+        });
+
+        this.testCase({
+            name: "SessionStorageSendBuffer deepCopy: function deepCopy() can return expected sessionStorage buffer",
+            test: () => {
+                let config = {
+                    endpointUrl: "https//: test",
+                    emitLineDelimitedJson: false,
+                    maxBatchInterval: 15000,
+                    maxBatchSizeInBytes: 102400,
+                    disableTelemetry: false,
+                    enableSessionStorageBuffer: true,
+                    isRetryDisabled: false,
+                    isBeaconApiDisabled:true,
+                    disableXhr: false,
+                    onunloadDisableFetch: false,
+                    onunloadDisableBeacon: false,
+                    instrumentationKey:"key",
+                    namePrefix: "",
+                    samplingPercentage: 100,
+                    customHeaders: [{header:"header",value:"val" }],
+                    convertUndefined: "",
+                    eventsLimitInMem: 10000
+                } as ISenderConfig;
+                let logger = new DiagnosticLogger({instrumentationKey: "abc"});
+                
+                let sessionBuffer = new SessionStorageSendBuffer(logger, config);
+                let sessionBufferCopy = new SessionStorageSendBuffer(logger, config);
+             
+                sessionBuffer.deepCopy(sessionBufferCopy);
+                QUnit.assert.deepEqual(sessionBufferCopy.getItems(), [], "payload should be empty");
+                QUnit.assert.deepEqual(this._getBuffer(BUFFER_KEY, logger), [], "session storage buffer should be empty");
+                let payload = ["payload1", "payload2", "payload3", "payload4", "payload5", "payload6"];
+                arrForEach(payload, (val) =>{
+                    sessionBuffer.enqueue(val);
+                });
+                QUnit.assert.deepEqual(this._getBuffer(BUFFER_KEY, logger), payload, "session storage buffer is set");
```

Comment:
It should also include items "marked as sent" as that is really where we have a risk of loosing events in this scenario